### ISSUE TITLE
Render all string properties of a dynamic schema as TextArea instead of Inputs

### ISSUE THREAD

**Issue by MaximeSauvageD4** [rjsf-team/react-jsonschema-form#3562]
### Prerequisites

- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)

### What theme are you using?

antd

### What is your question?

Hello team 🙂 

My App received dynamic schemas and thanks to your lib I'm able to generate corresponding forms. 
But when it comes to properties with type "string", all the inputs are rendered as basic inputs. 
It goes well for properties like "Title" when you don't expect a lot of characters.
But fo properties such as "Decription" it's not very user friendly to have this big amount of characters displayed in a single line (and having to scroll horizontally to see your content).
Because my App received dynamic JSON schemas, I'm not able to target specific properties with a ui:schema and tell them to render as TextArea. I don't know the properties in advance.

So I've read the documentation and I see about Widgets and Templates. 

I already use some templates such as ErrorListTemplate, ArrayFieldTemplate, ObjectFieldTemplate, FieldTemplate, FieldErrorTemplate to modify the original position of rendered elements.
I've also seen that you can override the default [input](https://rjsf-team.github.io/react-jsonschema-form/docs/advanced-customization/custom-templates#baseinputtemplate) which is wrapped by TextWidget.

My first option was to override this BaseInputTemplate but I see some disadvantages of doing so:
- having to set conditions to render a TextArea only if the field type is a string but what about enums or other special cases?
- having to redo all the connections with ANTD Input and ANTD TextArea to make things work again

Is there an easy way to specify to the lib that instead of using TextWidget, it should always use TextareaWidget or should I rewrite all the logic by overriding the BaseInputTemplate?

Thanks for your replies!

**Comment by heath-freenome** [comment]
You could update the `widgets` to make `TextWidget` be `TextAreaWidget` something like this:
```
import `generateWidgets` from `@rjsf/antd`;

const widgets = generateWidgets();

widgets.TextWidget = widgets.TextareaWidget;

<Form ... widgets={widgets} />
```

**Comment by MaximeSauvageD4** [comment]
Hi @heath-freenome,

Thanks for your reply! Your solution changed the traditional text inputs to textarea which is great, but it also changed the input type number to textarea which caused the arrows to increase and decrease the value to disappear... Also reprenseting a number in a huge textarea may be a little bit confusing. Otherwise, enums are still represented as select.

Unfortunately this solution does not fit my needs but I'm now aware of this and it could help us for a future issue.

What I end up with is to check the implementation of the BasicInputTemplate for ANTD (https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/antd/src/templates/BaseInputTemplate/index.tsx), taking this file, and modifying the Input tag to an Input.Textarea tag (with an autosize props for a better rendering).

I'm a little bit afraid of future modifications of BasicInputTeplate file that I will probably missing but for now, it's the only working solution with minimum effort for my needs.

Can we consider implementing an option in the future to make only text inputs to render as textarea inputs?

Thanks 🙂

**Comment by heath-freenome** [comment]
@MaximeSauvageD4 That's right, I forgot about that. Sorry it didn't work out. Well your solution is probably better anyway. In my own world, I've replaced the BaseInputTemplate myself. Hopefully if there are any changes that are made you can pick them up from a simple diff. I'm going to close this. Feel free to reopen if you need to.
